pipelines/build/regeneration/build_pipeline_generator.groovy

```diff
@@ -1,59 +1,143 @@
     }
-    
-    def config = [
-      TEST                : false,
-      GIT_URL             : "https://github.com/AdoptOpenJDK/openjdk-build.git",
-      BRANCH              : "master",
-      BUILD_FOLDER        : "build-scripts",
-      JOB_NAME            : "openjdk${javaVersion}-pipeline",
-      SCRIPT              : "pipelines/build/openjdk${javaVersion}_pipeline.groovy",
-      disableJob          : false,
-      triggerSchedule     : ""
-    ];
-    checkout([$class: 'GitSCM', userRemoteConfigs: [[url: config.GIT_URL]]])
-    
-    def target;
-    try {
-      target = load "${WORKSPACE}/pipelines/jobs/configurations/jdk${javaVersion}u.groovy"
-    } catch(Exception e) {
-      try {
-        target = load "${WORKSPACE}/pipelines/jobs/configurations/jdk${javaVersion}.groovy"
-      } catch(Exception e2) {
-        println "[WARNING] No config found for JDK${javaVersion}"
+
+    // Checkout into repository
+    checkout(
+      [
+        $class: 'GitSCM',
+        branches: [ [ name: repoBranch ] ],
+        userRemoteConfigs: [ remoteConfigs ]
+      ]
+    )
+
+    // Load jobRoot. This is where the openjdkxx-pipeline jobs will be created.
+    def jobRoot = (params.JOB_ROOT) ?: DEFAULTS_JSON["jenkinsDetails"]["rootDirectory"]
+
+    // Load scriptFolderPath. This is the folder where the openjdkxx-pipeline.groovy code is located compared to the repository root. These are the top level pipeline jobs.
+    def scriptFolderPath = (params.SCRIPT_FOLDER_PATH) ?: DEFAULTS_JSON["scriptDirectories"]["upstream"]
+
+    // Load nightlyFolderPath. This is the folder where the jdkxx.groovy code is located compared to the repository root. These define what the default set of nightlies will be.
+    def nightlyFolderPath = (params.NIGHTLY_FOLDER_PATH) ?: DEFAULTS_JSON["configDirectories"]["nightly"]
+
+    // Load jobTemplatePath. This is where the pipeline_job_template.groovy code is located compared to the repository root. This actually sets up the pipeline job using the parameters above.
+    def jobTemplatePath = (params.JOB_TEMPLATE_PATH) ?: DEFAULTS_JSON["jobTemplateDirectories"]["upstream"]
+
+    // Load enablePipelineSchedule. This determines whether we will be generating the pipelines with a schedule (defined in jdkxx.groovy) or not.
+    def enablePipelineSchedule = false
+    if (params.ENABLE_PIPELINE_SCHEDULE) {
+      enablePipelineSchedule = true
+    }
+
+    println "[INFO] Running generator script with the following configuration:"
+    println "REPOSITORY_URL = $repoUri"
+    println "REPOSITORY_BRANCH = $repoBranch"
+    println "JOB_ROOT = $jobRoot"
+    println "SCRIPT_FOLDER_PATH = $scriptFolderPath"
+    println "NIGHTLY_FOLDER_PATH = $nightlyFolderPath"
+    println "JOB_TEMPLATE_PATH = $jobTemplatePath"
+    println "ENABLE_PIPELINE_SCHEDULE = $ENABLE_PIPELINE_SCHEDULE"
+
+    (8..30).each({javaVersion ->
```

Comment:
8 to 30?

Reply:
It should really be [tip_version](https://api.adoptopenjdk.net/v3/info/available_releases)+1

Reply:
Done